msal-dotnet-articles/acquiring-tokens/web-apps-apis/on-behalf-of-flow.md

```diff
@@ -116,17 +116,28 @@ var authResult = await ((ILongRunningWebApi)confidentialClientApp)
 
 `userAccessToken` is a user access token used to call this web API. `sessionKey` will be used as a key when caching and retrieving the OBO token. If set to `null`, MSAL will set it to the assertion hash of the passed-in user token. It can also be set by the developer to something that identifies a specific user session, like the optional `sid` claim from the user token (for more information, see [Provide optional claims to your app](/azure/active-directory/develop/active-directory-optional-claims)). `InitiateLongRunningProcessInWebApi` doesn't check the cache; it will use the user token to acquire a new OBO token from Microsoft Entra ID, which will then be cached and returned.
 
-2. In the long-running process, whenever OBO token is needed, call:
+2. In the long-running process, whenever OBO token is needed, you call `AcquireTokenInLongRunningProcess` in the following pattern:
 
 ```csharp
-var authResult = await ((ILongRunningWebApi)confidentialClientApp)
-         .AcquireTokenInLongRunningProcess(
-              scopes,
-              sessionKey)
-         .ExecuteAsync();
+try {  
+    var authResult = await ((ILongRunningWebApi)confidentialClientApp)  
+         .AcquireTokenInLongRunningProcess(  
+              scopes,  
+              sessionKey)  
+         .ExecuteAsync();  
+} catch (MsalClientException) {  
+    // No tokens were found with this cache key.  
+    // First call InitiateLongRunningProcessInWebApi with a valid user assertion  
+    // to acquire tokens from Microsoft Entra ID and cache them.  
+} catch (MsalUiRequiredException) {  
+    // A refresh token was used to acquire new tokens  
```

Comment:
Too general purpose, might as well show how to create a WWW-Authenticate header? At least pseudocode / as a comment. 

Or point to Microsoft.Identity.Web product code that does this.

Reply:
Wouldnt claims already be in the exception in this case? So we can have them prompt the user to sign in and then try again.

Actually, it would be a `MsalClaimsChallengeException`` in this case

Reply:
How can a web api prompt a user? Only the client can prompt a user. The web api responds with `401 WWW-Authenticate Bearer claims={foo}` to indicate that a prompt is needed.

Reply:
Oh ok, I misread your comment. I can update